core/src/main/java/io/questdb/cutlass/text/ParallelCsvFileImporter.java

```diff
@@ -783,6 +792,91 @@ private int getTaskCount() {
         return taskDistribution.size() / 3;
     }
 
+    private void initWriterAndOverrideImportMetadata(
+            ObjList<CharSequence> names,
+            ObjList<TypeAdapter> types,
+            CairoSecurityContext cairoSecurityContext,
+            TypeManager typeManager
+    ) throws TextException {
+        final TableWriter writer = cairoEngine.getWriter(cairoSecurityContext, tableToken, LOCK_REASON);
+        final RecordMetadata metadata = GenericRecordMetadata.copyDense(writer.getMetadata());
+
+        if (metadata.getColumnCount() < types.size()) {
+            writer.close();
+            throw TextException.$("column count mismatch [textColumnCount=").put(types.size())
+                    .put(", tableColumnCount=").put(metadata.getColumnCount())
+                    .put(", table=").put(tableName)
+                    .put(']');
+        }
+
+        // remap index is only needed to adjust names and types
+        // workers will import data into temp tables without remapping
+        final IntList remapIndex = new IntList();
+        remapIndex.ensureCapacity(types.size());
+        for (int i = 0, n = types.size(); i < n; i++) {
+            final int columnIndex = metadata.getColumnIndexQuiet(names.getQuick(i));
+            final int idx = columnIndex > -1 ? columnIndex : i; // check for strict match ?
+            remapIndex.set(i, idx);
```

Comment:
Why you don't use `witerIndex` for the column here?

Reply:
That's because unlike in `TestLoader`, parallel importer uses remap index only to adjust names and types. It's not used to interact with `TableWriter`.